[pt-br] Localize docs/concepts/windows/intro (#48932)
kubernetes/website: content/pt-br/docs/concepts/windows/intro.md

    * `spec.containers[*].securityContext.runAsGroup`

    Na lista acima, curingas (`*`) indicam todos os elementos em uma lista. Por exemplo, `spec.containers[*].securityContext` refere*se ao objeto SecurityContext para todos os containers. Se qualquer um desses campos for especificado, o Pod não será admitido pelo servidor API.

* [Recursos de carga de trabalho](/pt-br/docs/concepts/workloads/controllers/) incluindo:
  * ReplicaSet
  * Deployment
  * StatefulSet
  * DaemonSet
  * Job
  * CronJob
  * ReplicationController
* {{< glossary_tooltip text="Services" term_id="service" >}}

  Veja [Balanceamento de carga e Services](/docs/concepts/services-networking/windows-networking/#load-balancing-and-services) para mais detalhes.

Pods, recursos de carga de trabalho e Services são elementos críticos para gerenciar cargas de trabalho Windows no Kubernetes. No entanto, por si só, eles não são suficientes para habilitar o gerenciamento adequado do ciclo de vida de cargas de trabalho Windows em um ambiente nativo da nuvem dinâmico.

* `kubectl exec`
* Métricas de Pod e container
* {{< glossary_tooltip text="Escalonamento horizontal de pods" term_id="horizontal-pod-autoscaler" >}}

> * {{< glossary_tooltip text="Quotas de recursos" term_id="resource*quota" >}}

Reviewer: Pequeno erro de digitação no nome do termo

```suggestion
* {{< glossary_tooltip text="Quotas de recursos" term_id="resource-quota" >}}
```

> * Pré-empção do schedule*

Reviewer: Acredito que a grafia correta seja:

```suggestion
* Preempção do scheduler
```

#### Compatibilidade de campos para especificações de container {#compatibility-v1-pod-spec-containers}

A lista a seguir documenta as diferenças entre como as especificações de container do Pod funcionam entre Windows e Linux:

* Huge pages não são implementadas no agente de execução de contêiner do Windows e não estão disponíveis. Elas requerem [afirmação de um privilégio de usuário](https://docs.microsoft.com/pt-br/windows/desktop/Memory/large-page-support) que não é configurável para contêineres.
* `requests.cpu` e `requests.memory` - as solicitações são subtraídas dos recursos disponíveis do nó, para que possam ser usadas para evitar o superprovisionamento de um nó. No entanto, elas não podem ser usadas para garantir recursos em um nó superprovisionado. Elas devem ser aplicadas a todos os contêineres como uma boa prática se o operador quiser evitar o superprovisionamento completamente.
* `securityContext.allowPrivilegeEscalation` - não é possível no Windows; nenhuma das capacidades está conectada
* `securityContext.capabilities` - capacidades POSIX não são implementadas no Windows

> * `securityContext.privileged` - o Windows não suporta contêineres privilegiados, use [HostProcess contêineres](/docs/tasks/configure-pod-container/create-hostprocess-pod/) em vez disso

Reviewer: Ajuste pequeno de gramática

```suggestion
* `securityContext.privileged` - o Windows não suporta contêineres privilegiados, use [contêineres HostProcess](/docs/tasks/configure-pod-container/create-hostprocess-pod/) em vez disso
```

* `volumes`
  * Se você definir um volume `emptyDir`, não pode definir sua fonte de volume para `memory`.
* Você não pode habilitar `mountPropagation` para montagens de volume, pois isso não é suportado no Windows.

#### Compatibilidade de campos para hostNetwork {#compatibility-v1-pod-spec-containers-hostnetwork}

{{< feature-state for_k8s_version="v1.26" state="alpha" >}}

O kubelet agora pode solicitar que pods em execução em nós Windows usem o namespace de rede do host em vez de criar um novo namespace de rede de pod. Para habilitar essa funcionalidade, passe `--feature-gates=WindowsHostNetwork=true` para o kubelet.

{{< note >}}
Esta funcionalidade requer um agente de execução de contêiner que suporte essa funcionalidade.
{{< /note >}}

#### Compatibilidade de campos para o contexto de segurança do Pod {#compatibility-v1-pod-spec-containers-securitycontext}

Apenas `securityContext.runAsNonRoot` e `securityContext.windowsOptions` dos campos [`securityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context) do Pod funcionam no Windows.

## Detector de problemas do nó

> O detector de problemas do nó (veja [Monitorando a integridade do nó](/docs/tasks/debug/debug-cluster/monitor-node-health/) tem suporte preliminar para Windows. Para mais informações, visite a [página do GitHub](https://github.com/kubernetes/node-problem-detector#windows) do projeto.

Reviewer: Faltou um parêntese

```suggestion
O detector de problemas do nó (veja [Monitorando a integridade do nó](/docs/tasks/debug/debug-cluster/monitor-node-health/)) tem suporte preliminar para Windows. Para mais informações, visite a [página do GitHub](https://github.com/kubernetes/node-problem-detector#windows) do projeto.
```

    - `spec.securityContext.sysctls`
    - `spec.shareProcessNamespace`
    - `spec.securityContext.runAsUser`
    - `spec.securityContext.runAsGroup`
    - `spec.securityContext.supplementalGroups`
    - `spec.containers[*].securityContext.seLinuxOptions`
    - `spec.containers[*].securityContext.seccompProfile`
    - `spec.containers[*].securityContext.capabilities`
    - `spec.containers[*].securityContext.readOnlyRootFilesystem`
    - `spec.containers[*].securityContext.privileged`
    - `spec.containers[*].securityContext.allowPrivilegeEscalation`
    - `spec.containers[*].securityContext.procMount`
    - `spec.containers[*].securityContext.runAsUser`
    - `spec.containers[*].securityContext.runAsGroup`

    Na lista acima, curingas (`*`) indicam todos os elementos em uma lista. Por exemplo, `spec.containers[*].securityContext` refere-se ao objeto SecurityContext para todos os containers. Se qualquer um desses campos for especificado, o Pod não será admitido pelo servidor API.

- **[Recursos de carga de trabalho](/pt-br/docs/concepts/workloads/controllers/)** incluindo:
  - ReplicaSet
  - Deployment
  - StatefulSet
  - DaemonSet
  - Job
  - CronJob
  - ReplicationController

> - **{{< glossary_tooltip text="Serviços" term_id="service" >}}**
> 
>   Veja [Balanceamento de carga e Serviços](/docs/concepts/services-networking/windows-networking/#load-balancing-and-services) para mais detalhes.

Reviewer: Algum motivo específico para o uso de destaque aqui? A página original não tem destaque neste texto.

Service aqui se refere à API do Kubernetes e não deve ser traduzido.

```suggestion
- {{< glossary_tooltip text="Services" term_id="service" >}}

  Veja [Balanceamento de carga e Services](/docs/concepts/services-networking/windows-networking/#load-balancing-and-services) para mais detalhes.
```

Pods, recursos de carga de trabalho e Services são elementos críticos para gerenciar cargas de trabalho Windows no Kubernetes. No entanto, por si só, eles não são suficientes para habilitar o gerenciamento adequado do ciclo de vida de cargas de trabalho Windows em um ambiente nativo da nuvem dinâmico.

- `kubectl exec`
- Métricas de Pod e container
- {{< glossary_tooltip text="Escalonamento horizontal de pods" term_id="horizontal-pod-autoscaler" >}}
- {{< glossary_tooltip text="Quotas de recursos" term_id="resource-quota" >}}
- Pré-empção do scheduler

### Opções de linha de comando para o kubelet {#compatibilidade-kubelet}

Algumas opções de linha de comando do kubelet se comportam de maneira diferente no Windows, conforme descrito abaixo:

> - A opção `--windows-priorityclass` permite definir a prioridade de agendamento do processo kubelet (veja [Gerenciamento de recursos de CPU](/pt-br/docs/concepts/configuration/windows-resource-management/)

Reviewer: Faltou fechar o parêntese

```suggestion
- A opção `--windows-priorityclass` permite definir a prioridade de agendamento do processo kubelet (veja [Gerenciamento de recursos de CPU](/pt-br/docs/concepts/configuration/windows-resource-management/))
```

- **Permissões de arquivo**: o Windows usa uma lista de controle de acesso baseada em SIDs, enquanto sistemas POSIX como Linux usam uma máscara de bits baseada em permissões de objeto e UID+GID, além de listas de controle de acesso _opcionais_.
- **Caminhos de arquivo**: a convenção no Windows é usar `\` em vez de `/`. As bibliotecas Go IO normalmente aceitam ambos e simplesmente funcionam, mas quando você está definindo um caminho ou linha de comando que é interpretada dentro de um container, pode ser necessário usar `\`.
- **Sinais**: Aplicativos interativos do Windows lidam com a terminação de maneira diferente e podem implementar um ou mais destes:
  - Uma thread de interface do usuário manipula mensagens bem definidas, incluindo `WM_CLOSE`.
  - Aplicativos de console lidam com Ctrl-C ou Ctrl-break usando um Manipulador de Controle.

>   - Serviços registram uma função Manipuladora de Controle de Serviço que pode aceitar códigos de controle `SERVICE_CONTROL_STOP`.

Reviewer: Estes itens não são destacados no texto original. Existe algum motivo para modificar a formatação do texto aqui?

Códigos de saída de container seguem a mesma convenção onde 0 é sucesso e diferente de zero é falha. Os códigos de erro específicos podem diferir entre Windows e Linux. No entanto, códigos de saída passados dos componentes do Kubernetes (kubelet, kube-proxy) são inalterados.

#### Compatibilidade de campos para especificações de container {#compatibilidade-v1-pod-spec-containers}

A lista a seguir documenta as diferenças entre como as especificações de container do Pod funcionam entre Windows e Linux:

- **Huge pages** não são implementadas no runtime de container do Windows e não estão disponíveis. Elas requerem [afirmação de um privilégio de usuário](https://docs.microsoft.com/pt-br/windows/desktop/Memory/large-page-support) que não é configurável para containers.

> - `requests.cpu` e `requests.memory` - as solicitações são subtraídas dos recursos disponíveis do nó, para que possam ser usadas para evitar o overprovisioning de um nó. No entanto, elas não podem ser usadas para garantir recursos em um nó superprovisionado. Elas devem ser aplicadas a todos os containers como uma boa prática se o operador quiser evitar o overprovisioning completamente.

Reviewer: Superprovisionado já foi usado no mesmo parágrafo -- algum motivo para não usar superprovisionamento?

```suggestion
- `requests.cpu` e `requests.memory` - as solicitações são subtraídas dos recursos disponíveis do nó, para que possam ser usadas para evitar o superprovisionamento de um nó. No entanto, elas não podem ser usadas para garantir recursos em um nó superprovisionado. Elas devem ser aplicadas a todos os containers como uma boa prática se o operador quiser evitar o superprovisionamento completamente.
```

- `volumes`
  - Se você definir um volume `emptyDir`, não pode definir sua fonte de volume para `memory`.
- Você não pode habilitar `mountPropagation` para montagens de volume, pois isso não é suportado no Windows.

#### Compatibilidade de campos para hostNetwork {#compatibilidade-v1-pod-spec-containers-hostnetwork}

{{< feature-state for_k8s_version="v1.26" state="alpha" >}}

O kubelet agora pode solicitar que pods em execução em nós Windows usem o namespace de rede do host em vez de criar um novo namespace de rede de pod. Para habilitar essa funcionalidade, passe `--feature-gates=WindowsHostNetwork=true` para o kubelet.

{{< note >}}
Esta funcionalidade requer um runtime de container que suporte essa funcionalidade.
{{< /note >}}

#### Compatibilidade de campos para o contexto de segurança do Pod {#compatibilidade-v1-pod-spec-containers-securitycontext}

Apenas `securityContext.runAsNonRoot` e `securityContext.windowsOptions` dos campos [`securityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context) do Pod funcionam no Windows.

## Detector de problemas do nó

> O detector de problemas do nó (veja [Monitorando a saúde do nó](/docs/tasks/debug/debug-cluster/monitor-node-health/) tem suporte preliminar para Windows. Para mais informações, visite a [página do GitHub](https://github.com/kubernetes/node-problem-detector#windows) do projeto.

Reviewer: Geralmente usamos integridade ao invés de saúde como tradução de health (já que não se trata de sistema orgânico). Também faltou fechar um parêntese aqui.

```suggestion
O detector de problemas do nó (veja [Monitorando a integridade do nó](/docs/tasks/debug/debug-cluster/monitor-node-health/)) tem suporte preliminar para Windows. Para mais informações, visite a [página do GitHub](https://github.com/kubernetes/node-problem-detector#windows) do projeto.
```

## Runtimes de container {#container-runtime}

Você precisa instalar um {{< glossary_tooltip text="runtime de container" term_id="container-runtime" >}} em cada nó do cluster para que os Pods possam ser executados lá.

Os seguintes runtimes de container funcionam com Windows:

{{% thirdparty-content %}}

### ContainerD

{{< feature-state for_k8s_version="v1.20" state="stable" >}}

Você pode usar {{< glossary_tooltip term_id="containerd" text="ContainerD" >}} 1.4.0+ como o runtime de container para nós Kubernetes que executam Windows.

Aprenda como [instalar o ContainerD em um nó Windows](/docs/setup/production-environment/container-runtimes/#containerd).

{{< note >}}

> Há uma [limitação conhecida](/pt-br/docs/tasks/configure-pod-container/configure-gmsa/s) ao usar GMSA com containerd para acessar compartilhamentos de rede do Windows, o que requer um patch no kernel.

Reviewer: Este link está quebrado.

{{< /note >}}

### Mirantis Container Runtime {#mcr}

[Mirantis Container Runtime](https://docs.mirantis.com/mcr/20.10/overview.html) (MCR) está disponível como um runtime de container para todas as versões do Windows Server 2019 e posteriores.

Veja [Instalar MCR em servidores Windows](https://docs.mirantis.com/mcr/20.10/install/mcr-windows.html) para mais informações.

> ## Compatibilidade de versão do Windows OS {#windows-os-versãos-suporte}

Reviewer: Esta âncora precisa ser mantida no original para que o link do início da página funcione.

```suggestion
## Compatibilidade de versão do Windows OS {#windows-os-version-support}
```

## Obtendo ajuda e solucionando problemas {#troubleshooting}

Sua principal fonte de ajuda para solucionar problemas em seu cluster Kubernetes deve começar com a página de [Solução de Problemas](/docs/tasks/debug/).

Alguma ajuda adicional, específica para Windows, está incluída nesta seção. Logs são um elemento importante na solução de problemas no Kubernetes. Certifique-se de incluí-los sempre que buscar assistência de outros colaboradores. Siga as instruções no [guia de contribuição do SIG Windows sobre coleta de logs](https://github.com/kubernetes/community/blob/master/sig-windows/CONTRIBUTING.md#gathering-logs).

> ### Relatando problemas e solicitações de recursos

Reviewer: Sugestão: feature aqui se refere a uma implementação de nova funcionalidade no software já existente.

```suggestion
### Relatando problemas e solicitações de funcionalidades
```

> Se você tiver algo que pareça um bug ou gostaria de fazer uma solicitação de recurso, siga o [guia de contribuição do SIG Windows](https://github.com/kubernetes/community/blob/master/sig-windows/CONTRIBUTING.md#reporting-issues-and-feature-requests) para criar um novo problema. Você deve primeiro pesquisar na lista de problemas caso tenha sido relatado anteriormente e comentar com sua experiência no problema e adicionar logs adicionais. O canal SIG Windows no Slack do Kubernetes também é uma ótima maneira de obter suporte inicial e ideias de solução de problemas antes de criar um ticket.

Reviewer: Sugestão: não traduzir issue aqui, pois se trata da funcionalidade do GitHub.

```suggestion
Se você tiver algo que pareça um bug ou gostaria de fazer uma solicitação de recurso, siga o [guia de contribuição do SIG Windows](https://github.com/kubernetes/community/blob/master/sig-windows/CONTRIBUTING.md#reporting-issues-and-feature-requests) para criar uma nova issue. Você deve primeiro pesquisar na lista de issues existentes caso tenha sido relatado anteriormente e comentar com sua experiência na issue e adicionar logs adicionais. O canal SIG Windows no Slack do Kubernetes também é uma ótima maneira de obter suporte inicial e ideias de solução de problemas antes de criar um ticket.
```

### Validando a operabilidade do cluster Windows

> O projeto Kubernetes fornece uma especificação de _Readiness Operacional do Windows_, acompanhada por um conjunto de testes estruturados. Este conjunto é dividido em dois conjuntos de testes, core e extended, cada um contendo categorias destinadas a testar áreas específicas. Pode ser usado para validar todas as funcionalidades de um sistema Windows e híbrido (misturado com nós Linux) com cobertura total.

Reviewer: Como Windows Operational Readiness é o nome do projeto, seria melhor não traduzir.

Core e extended são palavras que possuem boas traduções no português:
```suggestion
O projeto Kubernetes fornece uma especificação de _Windows Operational Readiness_, acompanhada por um conjunto de testes estruturados. Este conjunto é dividido em dois conjuntos de testes, principal (_core_) e estendido (_extended_), cada um contendo categorias destinadas a testar áreas específicas. Pode ser usado para validar todas as funcionalidades de um sistema Windows e híbrido (misturado com nós Linux) com cobertura total.
```

Para configurar o projeto em um cluster recém-criado, consulte as instruções no [guia do projeto](https://github.com/kubernetes-sigs/windows-operational-readiness/blob/main/README.md).

## Ferramentas de implantação

> A ferramenta kubeadm ajuda você a implantar um cluster Kubernetes, fornecendo o plano de controle para gerenciá-lo, e nós para executar suas cargas de trabalho.

Reviewer: ```suggestion
A ferramenta kubeadm ajuda você a implantar um cluster Kubernetes, fornecendo a camada de gerenciamento para gerenciá-lo, e nós para executar suas cargas de trabalho.
```
